src/deadline/unreal_adaptor/UnrealClient/step_handlers/unreal_render_step_handler.py

```diff
@@ -132,6 +132,11 @@ def executor_failed_callback(executor, pipeline, is_fatal, error):
     def executor_finished_callback(movie_pipeline=None, results=None):
         unreal.log("Render Executor: Rendering is complete")
 
+    @staticmethod
+    def log(message: str):
+        if unreal:
```

Comment:
This will get us there!  I was kind of imagining we'd subclass logging.Handler and override emit to unreal.log, and only attach that handler when unreal was available (Like in init_unreal).  Just so I understand, was there a reason to prefer this way?

Reply:
the reason is where unreal logger wrapper lives. I think its not a good idea to create it in src/deadline to make available in adaptor and submitter since only in single render handler (inside adaptor) unreal logic used. And dont want to duplicate code in adaptor to implement the same logger logic. I think we want to follow the approach where only unreal_adaptor and unreal_submitter as independent packages live inside deadline folder

Reply:
Definitely doesn't make sense to duplicate code if we can avoid it, I agree!  But having shared code "somewhere" I don't think would be a bad thing.  Consistency across the codebase I think would be valuable long term. It's not a blocking issue to have it this way though